jdbc/runtime/src/main/java/io/quarkiverse/configextensions/jdbc/runtime/JdbcConfigSourceFactory.java

```diff
@@ -0,0 +1,72 @@
+package io.quarkiverse.configextensions.jdbc.runtime;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.jboss.logging.Logger;
+
+import io.smallrye.config.ConfigSourceContext;
+import io.smallrye.config.ConfigSourceFactory;
+import io.smallrye.config.common.MapBackedConfigSource;
+
+public class JdbcConfigSourceFactory implements ConfigSourceFactory {
+    private static final Logger log = Logger.getLogger(JdbcConfigSourceFactory.class);
+    Repository repository = null;
+
+    @Override
+    public Iterable<ConfigSource> getConfigSources(ConfigSourceContext context) {
+        return getConfigSource(context);
+    }
+
+    private List<ConfigSource> getConfigSource(ConfigSourceContext context) {
+
+        boolean enabled = Boolean.valueOf(
+                Optional.ofNullable(context.getValue("quarkus.config.source.jdbc.enabled").getValue()).orElse("true"));
+        if (!enabled) {
+            return Collections.emptyList();
+        }
+
+        String table = Optional.ofNullable(context.getValue("quarkus.config.source.jdbc.table").getValue())
+                .orElse("configuration");
+        String keyColumn = Optional.ofNullable(context.getValue("quarkus.config.source.jdbc.key").getValue())
```

Comment:
`2021-04-23 14:39:45,228 WARN  [io.qua.config] (Quarkus Main Thread) Unrecognized configuration key "quarkus.config.jdbc.enabled" was provided; it will be ignored; verify that the dependency extension for this configuration is set or you did not make a typo`

I think context.getValue() is not correctly analized by the configuration validator

Reply:
Yes, because the Quarkus config prefix validator runs after the Config is initialised and at this stage the Config is still initialising.

This reminds me that we probably should not be using the `quarkus` namespace for this. @gastaldi did we ever settle on which prefix should Quarkiverse extensions should use?

Reply:
@radcortez we had some discussion around this here: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Quarkiverse.20and.20config.20prefix/near/214184982

Reply:
Hum, it seems we didn't reach a conclusion. Let me restart that.

Reply:
Is there a conclusion yet?

Reply:
Ops, sorry for the delay. Yes, let's use the `quarkus` namespace. 

To trick the validation to pass, we need create a ConfigRoot. I've done the same here: 
https://github.com/quarkusio/quarkus/blob/d3d70ac5b97932bf2a2811ef796219d590ff2811/core/runtime/src/main/java/io/quarkus/runtime/ConfigConfig.java#L17

This will also provide documentation for the configuration.

Thanks!

Reply:
I've added the ConfigRoot class, do I have to do something else? The "Unrecognized configuration" error is still showing.

Reply:
Hi, I tried, but it keeps ignoring the Config. Can you point me to some code example?

Reply:
Ah, because the `JdbcConfigConfig` sets the name to `quarkus.config.source.jdbc`, but the `quarkus` prefix is added automatically, so quarkus is searching for `quarkus.quarkus.config.source.jdbc`. Please remove `quarkus` and keep only `config.source.jdbc`.

Reply:
Thanks, all clear now!